lms/djangoapps/learner_home/serializers.py

```diff
@@ -228,10 +229,14 @@ def get_canUpgrade(self, enrollment):
         )
 
     def get_isAuditAccessExpired(self, enrollment):
-        show_courseware_link = self.context.get("show_courseware_link", {}).get(
-            enrollment.course.id, {}
+        """Mirrors logic in "check_course_expired" but using pre-fetched expiration date"""
+        expiration_date = self.context.get("audit_access_deadlines", {}).get(
+            enrollment.course_id
         )
-        return show_courseware_link.get("error_code") == "audit_expired"
+
+        if expiration_date and timezone.now() > expiration_date:
```

Comment:
`return expiration_date and timezone.now() > expiration_date` 😉

Reply:
Yup, fair point

Reply:
Slight caveat, this returns `None` if `expiration_date` is `None`. Should be:

```suggestion
        return bool(expiration_date) and timezone.now() > expiration_date:
```